dashboard/app/models/script.rb

```diff
@@ -605,9 +605,13 @@ def csf_finish_url
     end
   end
 
-  def summarize
+  def summarize(user)
```

Comment:
That makes sense. I think this means what I'll have to do is start out by hiding all lockable stages, then upon getting user progress (something we do asynchronously) unhide the lockable stages as appropriate. This might be kind of weird having new stages suddenly appear, but perhaps not any weirder than having progress appear asynchronously.